src/app/Main/Body/QueryPage/Results/TableResults/TableResults-Options.component.tsx

```diff
@@ -73,22 +61,17 @@ const BlueRadio = withStyles(() =>
  * @returns {JSX.Element} - option bar component
  */
 export function ResultsTableOptionsComponent(props: {
-    isSumRowCol: { rowSum: boolean; colSum: boolean };
+    isSumRowCol: { showRowSum: boolean; showColSum: boolean };
     tableOption: DisplayOptionType;
     onDisplayOptionsChange: (displayOption: string) => void;
-    onShowRowSumToggle: (showSums: boolean) => void;
-    onShowColSumToggle: (showSums: boolean) => void;
+    onShowRowColSum: (name: string, checked: boolean) => void;
 }): JSX.Element {
     const { t } = useTranslation(["QueryPage"]);
     const handleChangeDisplayOptions = (displayOption: string): void =>
         props.onDisplayOptionsChange(displayOption);
 
-    const handleChangeShowRowSumToggle = (showSums: boolean): void => {
-        props.onShowRowSumToggle(showSums);
-    };
-    const handleChangeShowColSumToggle = (showSums: boolean): void => {
-        props.onShowColSumToggle(showSums);
-    };
+    const handleChangeShowRowColSum = (name: string, checked: boolean): void =>
+        props.onShowRowColSum(name, checked);

```

Comment:
Same here, it is not a good idea to expose the CheckboxesComponent specific interface to the components up the tree. Here would be the place to map it to the options context. If you do so you have to track the sum options state here too.

Reply:
how should I do this exactly? Check if name === "showRowSum" and then assign the "checked" value? The function only ever gets the info for showRowSum or showColSum, not both at once.

Reply:
That's why you would need a state here too, to keep track of both checkboxes states. It would be the same pattern you used in the "QueryPageContent-TableResultsLayout" component. So you would apply the change of one checkbox to the state and emit it at a whole to the optionsChange event.

Reply:
And the handlers in the "QueryPageContent-TableResultsLayout" component would just copy the state.

Reply:
Done, I think... Commit: "Use SumOptions-interface"